### PR TITLE
Resource manager/watch type

### PR DESCRIPTION
**Pull request by CabinfeverB**
Add watch type `plan`. And modify last_duration_ms as int64 because of `time.Duration` is int64 and we should use negative to indicate `unlimited`.

### CODE REVIEW

File: proto/resource_manager.proto [hunk 1]
```diff
@@ -196,6 +196,7 @@ enum RunawayAction {
 enum RunawayWatchType {
   Exact = 0;
   Similar = 1;
+  Plan = 3;
```
Comment (glorv) [inline proto/resource_manager.proto:199]: If I understand correctly, in the current impl, Similar actually means Plan, so after this change Similar will mean SQL digest, right? So this is a incompatible change

Comment (CabinfeverB) [reply proto/resource_manager.proto:199]: Yes, we should also modify the docs

**Review by glorv — APPROVED** [review]
LGTM

File: proto/resource_manager.proto [hunk 2]
```diff
@@ -204,7 +205,7 @@ message RunawayRule {
 
 message RunawayWatch {
   // how long would the watch last
-  uint64 lasting_duration_ms = 1;
+  int64 lasting_duration_ms = 1;
```
Comment (Connor1996) [inline proto/resource_manager.proto:208]: why not use 0 as unlimited

Comment (CabinfeverB) [reply proto/resource_manager.proto:208]: In TiDB, ResourceGroupRunawaySettings is defined as follow,
```
type ResourceGroupRunawaySettings struct {
	ExecElapsedTimeMs uint64            `json:"exec_elapsed_time_ms"`
	Action            RunawayActionType `json:"action"`
	WatchType         RunawayWatchType  `json:"watch_type"`
	WatchDurationMs   int64             `json:"watch_duration_ms"`
}
```
I should use WatchDurationMs to check whether `Watch` is defined.
But I can use a pointer for `Watch`.   Which do you think is better

Comment (Connor1996) [reply proto/resource_manager.proto:208]: maybe add a none watch type?

Comment (glorv) [reply proto/resource_manager.proto:208]: +1

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**